simple-acquaintances/data/datagen.py

```diff
@@ -0,0 +1,400 @@
+"""
+Generates data for the simple-acquaintances PSL example.
+
+Use -h/--help for usage options.
+"""
+
+import argparse
+import json
+import numpy as np
+import os
+
+from itertools import combinations
+from scipy.stats import truncnorm
+from sklearn.metrics.pairwise import cosine_similarity
+
+DEFAULT_GLOBAL_LIKES = 5
+DEFAULT_LOCAL_LIKES = 5
+DEFAULT_LOCAL_LIKES_VARIANCE = 1.0
+DEFAULT_MAX_RAND_INT = 1000000
+DEFAULT_PEOPLE = 25
+DEFAULT_PLACES = 5
+DEFAULT_PLACES_LIVED_MEAN = 3.0
+DEFAULT_PLACES_LIVED_SD = 1.0
+DEFAULT_TARGET_SPLIT = 0.2
+
+
+def checkNonNegativeInt(arg):
+    """
+    Check if argument is non negative integer.
+    """
+    if not arg.isdigit():
+        raise argparse.ArgumentTypeError(
+            '{} is not a non negative integer.'.format(arg))
+    return int(arg)
+
+
+def checkValidSplit(arg):
+    """
+    Check if argument is a valid split for targets i.e {0...1}.
+    """
+    try:
+        if 0 <= float(arg) <= 1:
+            return arg
+        else:
+            raise argparse.ArgumentTypeError('The input should be a valid float between [0, 1].')
+    except ValueError as e:
```

Comment:
noted. Also, the method above works because the argument is read as a string from the command line. Thus if the user inputs something like "--globalLikes=-2" the statement `arg.isdigit()` returns `false`. It only returns `true`  if the input is [0, inf].

Reply:
I know that's just a language idiosyncrasy and that the code isn't obvious unless you know this specific detail but I didn't want to cast to an int and then work with it. If you feel that's a better way to go about it I'll change it!

Reply:
Don't reply to two different comments in a single thread!

First comment:
Although it sounds like the method will work, your response is strange.
Make sure you understand the situation.

Second comment:
That is not what I said at all!
Also, I think you are responding to the wrong comment.
Do you mean the `float(arg)` comment?
```
0.0 <= float(arg) <= 1.0
```

Reply:
First comment:
Comment this so people are not confused (as I was) by Python bad naming.